File: scratch/test_pagination_api.js
const API_BASE = 'http://localhost:3000';

async function testApi() {
    try {
        console.log("Testing /api/medicines with limit=10 and offset=0...");
        let res = await fetch(`${API_BASE}/api/medicines?limit=10&offset=0&hospital_name=sai%20hospital`);
        console.log(`Status: ${res.status}`);
        console.log("Headers:");
        for (const [key, val] of res.headers.entries()) {
            console.log(`  ${key}: ${val}`);
        }
        let data = await res.json();
        console.log(`Returned items: ${data.length}`);
        if (data.length > 0) {
            console.log("First item sample:", data[0]);
        }

        console.log("\nTesting /api/medicines with limit=10 and offset=10...");
        res = await fetch(`${API_BASE}/api/medicines?limit=10&offset=10&hospital_name=sai%20hospital`);
        console.log(`Status: ${res.status}`);
        data = await res.json();
        console.log(`Returned items: ${data.length}`);
        if (data.length > 0) {
            console.log("First item sample (page 2):", data[0]);
        }

        console.log("\nTesting search query on medicines...");
        res = await fetch(`${API_BASE}/api/medicines?limit=10&offset=0&search=Amox&hospital_name=sai%20hospital`);
        console.log(`Status: ${res.status}`);
        console.log(`X-Total-Count: ${res.headers.get('X-Total-Count')}`);
        data = await res.json();
        console.log(`Search results for 'Amox': ${data.length}`);
        if (data.length > 0) {
            console.log("First search item:", data[0]);
        }
    } catch (err) {
        console.error(err);
    }
}

testApi();

File: scratch/test_pagination_api.py
import requests  # type: ignore

API_BASE = 'http://localhost:3000'

def test_api():
    print("Testing /api/medicines with limit=10 and offset=0...")
    res = requests.get(f"{API_BASE}/api/medicines", params={
        'limit': 10,
        'offset': 0,
        'hospital_name': 'sai hospital'
    })
    print(f"Status: {res.status_code}")
    print(f"Headers: {dict(res.headers)}")
    data = res.json()
    print(f"Returned items: {len(data)}")
    if data:
        print("First item sample:")
        print(data[0])

    print("\nTesting /api/medicines with limit=10 and offset=10...")
    res2 = requests.get(f"{API_BASE}/api/medicines", params={
        'limit': 10,
        'offset': 10,
        'hospital_name': 'sai hospital'
    })
    data2 = res2.json()
    print(f"Returned items: {len(data2)}")
    if data2:
        print("First item sample (page 2):")
        print(data2[0])

    print("\nTesting search query on medicines...")
    res3 = requests.get(f"{API_BASE}/api/medicines", params={
        'limit': 10,
        'offset': 0,
        'search': 'Amox',
        'hospital_name': 'sai hospital'
    })
    data3 = res3.json()
    print(f"Search results for 'Amox': {len(data3)}")
    if data3:
        print("First search item:")
        print(data3[0])

if __name__ == '__main__':
    test_api()
